lib/reek/code_comment.rb

```diff
@@ -1,39 +1,45 @@
 require 'yaml'
 require 'private_attr/everywhere'
 
-# NOTE: Work-around for https://github.com/tenderlove/psych/issues/223
-require 'psych.rb' if Object.const_defined?(:Psych)
-
 module Reek
   #
   # A comment header from an abstract syntax tree; found directly above
   # module, class and method definitions.
   #
   class CodeComment
-    CONFIG_REGEX = /:reek:(\w+)(:\s*\{.*?\})?/
-
-    def initialize(text)
-      @config = Hash.new { |hash, key| hash[key] = {} }
-      @text = text.gsub(CONFIG_REGEX) do
-        @config.merge! add_to_config($1, $2)
-        ''
-      end.gsub(/#/, '').gsub(/\n/, '').strip
-    end
+    CONFIGURATION_REGEX            = /:reek:(\w+)(:\s*\{.*?\})?/
+    SANITIZE_REGEX                 = /#|\n|\s{2,}/ # Matches '#', newlines and > 1 whitespaces.
+    DISABLE_DETECTOR_CONFIGURATION = ': { enabled: false }'
+    MINIMUM_CONTENT_LENGTH         = 2
 
     attr_reader :config
+    private_attr_reader :original_comment
+
+    #
+    # @param comment [String] - the original comment as found in the source code
+    #   E.g.:
+    #   "\n        # :reek:Duplication: { enabled: false }\n      "
+    #
+    def initialize(comment)
+      @original_comment  = comment
+      @config            = Hash.new { |hash, key| hash[key] = {} }
+
+      @original_comment.scan(CONFIGURATION_REGEX) do |smell_type, options|
+        @config.merge! YAML.load(smell_type + (options || DISABLE_DETECTOR_CONFIGURATION))
+      end
+    end
 
     def descriptive?
-      text.split(/\s+/).length >= 2
+      sanitized_comment.split(/\s+/).length >= MINIMUM_CONTENT_LENGTH
     end
 
     private
 
-    # :reek:UtilityFunction
-    def add_to_config(smell, options)
-      options ||= ': { enabled: false }'
-      YAML.load(smell + options)
+    def sanitized_comment
+      @sanitized_comment ||= original_comment.
+        gsub(CONFIGURATION_REGEX, '').
+        gsub(SANITIZE_REGEX, '').
```

Comment:
I think this will convert `foo  bar` (with two spaces) to `foobar`.

Perhaps `gsub(/(#|\n|\s)+/, ' ') will to the right thing?